benchmarks/cugraph/standalone/cugraph_graph_creation.py

```diff
@@ -0,0 +1,182 @@
+from time import sleep
+import pandas as pd
+
+
+@get_allocation_counts_dask_persist(return_allocations=True, logging=False)
+def construct_graph(dask_dataframe, directed=False, renumber=False):
+    """
+    Args:
+        dask_dataframe:
+            dask_dataframe contains weighted and undirected edges with self
+            loops. Multiple edges will likely be present as well.
+        directed:
+            If True, the graph will be directed.
+        renumber:
+            If True, the graph will be renumbered.
+    Returns:
+        G:  cugraph.Graph
+    """
+    G = cugraph.Graph(directed=directed)
+    G.from_dask_cudf_edgelist(
+        dask_dataframe, source="src", destination="dst", renumber=renumber
+    )
+    return G
+
+
+def benchmark_cugraph_graph_creation(scale, edgefactor, seed, directed, renumber):
+    """
+    Entry point for the benchmark.
+    """
+    dask_df = generate_edgelist(
+        scale=scale, edgefactor=edgefactor, seed=seed, unweighted=True
+    )
+    dask_df = dask_df.astype("int64")
+    dask_df = dask_df.reset_index(drop=True)
+    input_memory = dask_df.memory_usage().sum().compute()
+    num_input_edges = len(dask_df)
+    print(
+        f"Number of input edges = {num_input_edges:,}, directed = {directed}, renumber = {renumber}"
+    )
+    G, allocation_counts = construct_graph(
+        dask_df, directed=directed, renumber=renumber
+    )
+    (
+        input_to_peak_ratio,
+        output_to_peak_ratio,
+        input_memory_per_worker,
+        peak_allocation_across_workers,
+    ) = get_memory_statistics(
+        allocation_counts=allocation_counts, input_memory=input_memory
+    )
+    print(f"Number of edges in final graph = {G.number_of_edges():,}")
+    print("-" * 80)
+    return (
+        num_input_edges,
+        input_to_peak_ratio,
+        output_to_peak_ratio,
+        input_memory_per_worker,
+        peak_allocation_across_workers,
+    )
+
+
+def get_memory_statistics(allocation_counts, input_memory):
+    """
+    Get memory statistics for the benchmark.
+    """
+    output_to_peak_ratio = get_peak_output_ratio_across_workers(allocation_counts)
+    peak_allocation_across_workers = max(
+        [a["peak_bytes"] for a in allocation_counts.values()]
+    )
+    input_memory_per_worker = input_memory / len(allocation_counts.keys())
+    input_to_peak_ratio = peak_allocation_across_workers / input_memory_per_worker
+    print(f"Edge List Memory = {sizeof_fmt(input_memory_per_worker)}")
+    print(f"Peak Memory across workers = {sizeof_fmt(peak_allocation_across_workers)}")
+    print(f"Max Peak to output graph ratio across workers = {output_to_peak_ratio:.2f}")
+    print(
+        f"Max Peak to avg input graph ratio across workers = {input_to_peak_ratio:.2f}"
+    )
+    return (
+        input_to_peak_ratio,
+        output_to_peak_ratio,
+        input_memory_per_worker,
+        peak_allocation_across_workers,
+    )
+
+
+def restart_client(client):
```

Comment:
I don't have a strong opinion, but I wonder if this should go in `testing.mg_utils` too.

Reply:
Yup, I should move it there.

Reply:
Resolved by https://github.com/rapidsai/cugraph/commit/005b698413ad4f3b849461831175ec75f04533f1